### ISSUE TITLE
Kernel: Cannot use full physical memory

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#4730]
If there's only about a couple hundred megabytes left, the system will suddenly start declaring that it's out of memory. This causes a bunch of weird effects. Best way to do this to start the system and run `allocate 285` then play around.

Demo:
![Peek 2021-01-01 23-46](https://user-images.githubusercontent.com/25595356/103448055-b2de2480-4c8b-11eb-9dd9-ceb167d8333c.gif)

**Comment by tomuta** [comment]
Yeah that's kind of expected now. The system now commits memory without actually allocating it. Unlike previously, where it allowed you to semi-allocate huge amounts and then potentially not be able to fulfill that promise. A lot of the system memory is already committed to be used by other processes, so you can't really allocate as much anymore. The reason other stuff crashes is largely ProcFS's fault, I have some other changes that I need to bring up to date that should help with that.
